core/commands/patch.go

```diff
@@ -0,0 +1,132 @@
+package commands
+
+import (
+	"fmt"
+	"io"
+	"strings"
+	"time"
+
+	mh "github.com/ipfs/go-ipfs/Godeps/_workspace/src/github.com/jbenet/go-multihash"
+	context "github.com/ipfs/go-ipfs/Godeps/_workspace/src/golang.org/x/net/context"
+
+	cmds "github.com/ipfs/go-ipfs/commands"
+	u "github.com/ipfs/go-ipfs/util"
+)
+
+var PatchCmd = &cmds.Command{
```

Comment:
this patch command is specific to objects, i'd put it inside `ipfs object patch`, not a whole porcelain command.